lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenLemminxExtension.java

```diff
@@ -765,12 +784,29 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 	/**
 	 * Returns the list of Maven Projects currently added to the Workspace
 	 * 
+	 * @param buildIfNecessary A boolean 'true' indicates that all projects are to
+	 *                         be returned, not only the cached ones at the moment,
+	 *                         method should wait for the final build result,
+	 *                         otherwise the only project that are already built and
+	 *                         cached are to be returned, the rest of the projects
+	 *                         are to be built in background
 	 * @return List of Maven Projects
 	 */
-	public List<MavenProject> getCurrentWorkspaceProjects() {
+	public List<MavenProject> getCurrentWorkspaceProjects(boolean wait) {
 		return workspaceReader.getCurrentWorkspaceArtifactFiles().stream()
-			.map(f -> getProjectCache().getLastSuccessfulMavenProject(f))
-			.filter(Objects::nonNull).toList();
+					.map(file -> {
+						try {
+							CompletableFuture<LoadedMavenProject> loadedProject = 
+									 getProjectCache().getLoadedMavenProject(toUriASCIIString(file));
+							return wait ? loadedProject.get() : loadedProject.getNow(null);

```

Comment:
Please don't never use .get(). us it with a timeout if it is required, but ideally, you shoudl return 

CompletabmeFuture<List<MavenProject>> getCurrentWorkspaceProjects

Reply:
The argument `wait == true` is used in Rename participant  - when renaming a Maven Property we need to get the full list of child maven projects of the project where the maven property is defined. Currently this support needs a synchronous way of getting such a children list.
In future the rename support can be modified to use  a ` CompletabmeFuture<List>` or other asynchronous functionality. 
I suggest reporting it in a separate issue.

Reply:
I think it should be safer to use a timeout to avoid blocking thread.

I prefer not having a rename feature than it blocks several threads and the editing support will be slow.

Reply:
Added a 10 seconds timeout - do you think it's OK?  (Testing in my local environment is successful with 5 seconds timeout value, but Jenkins CI build is about 1.5 times slower and Windows CI build is about 2 times slower than building in mi local environment, so I'd prefer to leave it that big for now)